File: cs/sample/Sample.Tests/BinarySearch/testBinarySearch.cs
using System;
using Xunit;
using Sample.BinarySearch;

namespace Sample.Tests{
    public class UnitTest_BinarySearch{
        [Theory]
        [InlineData(new int[] {-1,0,3,5,9,12},9,4)]
        [InlineData(new int[] {-1,0,3,5,9,12},2,-1)]
        public void checkResult(int[] nums,int target,int expected){
            Assert.Equal(expected,binarySearch.naive(nums,target));
        }
    }
}

File: py/tests/testBinarySearch/test_BinarySearch.py

import unittest
from parameterized import parameterized as p
from solns.binarySearch.binarySearch import *

class UnitTest_BinarySearch(unittest.TestCase):
    @p.expand([
        [[-1,0,3,5,9,12],9,4],[[-1,0,3,5,9,12],2,-1]
    ])
    def test_naive(self,nums,target,expected):
        self.assertEqual(Solution.naive(nums,target),expected)
